pkg/client/oci/oci.go

```diff
@@ -354,9 +405,113 @@ func (r *Repo) Reload() error {
 	return errors.Errorf("reload method is not supported yet")
 }
 
-func (r *Repo) newDockerResolver() remotes.Resolver {
+// SetEntries sets the entries of a repo
+func (r *Repo) SetEntries(entries map[string][]string) {
+	r.entries = entries
+}
+
+// OciReference contains info about an oci reference
+type ociReference struct {
+	path string
+	tag  string
+}
+
+// ociReferenceExists checks if a given oci reference exists in the repository
+// OCI endpoints: https://github.com/opencontainers/distribution-spec/blob/main/spec.md#endpoints
+func ociReferenceExists(repoUrl *url.URL, ociRef, username, password string, insecure bool) (bool, error) {
+	ociReference, err := name.ParseReference(ociRef)
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+	ociReferenceIdentifier := ociReference.Identifier()
+	parts := strings.Split(ociReference.Identifier(), ":")
+	if len(parts) == 2 { // Identifier is a digest. We should remove sha256 prefix
+		ociReferenceIdentifier = parts[1]
+	}
+	contextOciRef := fmt.Sprintf("%s://%s", repoUrl.Scheme, ociReference.Context())
+	u, err := url.Parse(contextOciRef)
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+	u.Path = path.Join("v2", u.Path, "manifests", ociReferenceIdentifier)
+
+	req, err := http.NewRequest("HEAD", u.String(), nil)
```

Comment:
as I mentioned in a previous comment. Have you tried to use https://pkg.go.dev/github.com/google/go-containerregistry@v0.7.0/pkg/v1/remote#Head ?

That one seems to require a `name.Referece` which you already have from line 422.

Reply:
I just tried it but there is something I don't like:

Currently I manage three situations based on exitCode:

- index exists. code 200
- index not exists. code 404
- other status are treated as error. For example wrong credentials

with remote.Head() method if the index does not exists directly throws an error:

```
Error: HEAD https://my-oci-registry.io/v2/my-project/charts/index/manifests/latest: unexpected status code 404 Not Found (HEAD responses have no body, use GET for details)
```

However, I don't want to throw an error if the index does not exist